app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadProgressScreen.kt

```diff
@@ -17,10 +17,236 @@
+        titleRes = R.string.uploadCancelConfirmBottomSheetTitle,
+        imageVector = AppIllus.RedCrossPaperPlanes.image(),
+        topButton = {
+            LargeButton(
+                titleRes = R.string.uploadCancelConfirmBottomSheetCancel,
+                modifier = it,
+                style = ButtonType.ERROR,
+                onClick = onCancel,
+            )
+        },
+        bottomButton = {
+            LargeButton(
+                titleRes = R.string.uploadCancelConfirmBottomSheetClose,
+                modifier = it,
+                style = ButtonType.TERTIARY,
+                onClick = closeButtonSheet,
+            )
+        },
+        onDismissRequest = closeButtonSheet,
+    )
+}
+
+@Composable
+private fun ColumnScope.WeightOneSpacer(minHeight: Dp) {
+    Spacer(modifier = Modifier.height(minHeight))
+    Spacer(modifier = Modifier.weight(1f))
+}
+
+@Composable
+fun Progress(uploadedSizeInBytes: () -> Long, totalSizeInBytes: () -> Long) {
+    Row {
+        Percentage(uploadedSizeInBytes, totalSizeInBytes)
+        Text(" - ")
+        UploadedSize(uploadedSizeInBytes)
+
+        TotalSize(totalSizeInBytes)
+    }
+}
+
+@Composable
+private fun Percentage(uploadedSizeInBytes: () -> Long, totalSizeInBytes: () -> Long) {
+    val percentageNoDecimals by remember {
+        derivedStateOf {
+            val percentage = (uploadedSizeInBytes().toFloat() / totalSizeInBytes())
+            String.format(Locale.getDefault(), "%d", (percentage * 100).toInt())
+        }
+    }
+
+    Text("$percentageNoDecimals%")
+}
+
+@Composable
+private fun UploadedSize(uploadedSizeInBytes: () -> Long) {
+    val context = LocalContext.current
+    val humanReadableSize by remember {
+        derivedStateOf { HumanReadableSizeUtils.getHumanReadableSize(context, uploadedSizeInBytes()) }
+    }
+
+    Text(humanReadableSize)
+}
+
+@Composable
+private fun TotalSize(totalSizeInBytes: () -> Long) {
+    val context = LocalContext.current
+    val humanReadableTotalSize by remember {
+        derivedStateOf { HumanReadableSizeUtils.getHumanReadableSize(context, totalSizeInBytes()) }
+    }
+
+    Text(" / $humanReadableTotalSize")
+}
+
+
+enum class UploadProgressAdType(
+    @StringRes private val descriptionTemplate: Int,
+    @StringRes private val descriptionAccentuatedPart: Int,
+    val illustration: ThemedImage,
+) {
+    INDEPENDENCE(
+        R.string.uploadSuccessDescriptionTemplateIndependence,
+        R.string.uploadSuccessDescriptionArgumentIndependence,
+        AppIllus.SwissWithFlag
+    ),
+    ENERGY(
+        R.string.uploadSuccessDescriptionTemplateEnergy,
+        R.string.uploadSuccessDescriptionArgumentEnergy,
+        AppIllus.MountainGondola
+    ),
+    CONFIDENTIALITY(
+        R.string.uploadSuccessDescriptionTemplateConfidentiality,
+        R.string.uploadSuccessDescriptionArgumentConfidentiality,
+        AppIllus.MetallicSafe
+    );
+
+    @Composable
+    fun description(): AnnotatedString {
```

Comment:
```suggestion
    fun Description(): AnnotatedString {
```

Reply:
I don't think we need to add an uppercase characters here. We usually use uppercase character for composable functions that represents a UI component

Reply:
You are correct : https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-api-guidelines.md#naming-composable-functions-that-return-values